src/ripple/app/tx/impl/CancelTicket.cpp

```diff
@@ -41,49 +42,23 @@ class CancelTicket
 
     }
 
-    TER doApply () override
+    TER preCheck () override
     {
-        assert (mTxnAccount);
-
-        uint256 const ticketId = mTxn.getFieldH256 (sfTicketID);
-
-        SLE::pointer sleTicket = mEngine->view ().entryCache (ltTICKET, ticketId);
-
-        if (!sleTicket)
-            return tecNO_ENTRY;
-
-        auto const ticket_owner =
-            sleTicket->getFieldAccount160 (sfAccount);
-
-        bool authorized =
-            mTxnAccountID == ticket_owner;
-
-        // The target can also always remove a ticket
-        if (!authorized && sleTicket->isFieldPresent (sfTarget))
-            authorized = (mTxnAccountID == sleTicket->getFieldAccount160 (sfTarget));
+        // A Ticket may not be used to cancel a Ticket.  So a Sequence of
+        // zero (which indicates that a Ticket is being used) is malformed.
+        if (mTxn.getSequence () == 0)
+            return tefPAST_SEQ;
```

Comment:
Returning `tefPAST_SEQ` is incorrect. a `tef` code should be returned when a transaction can't be applied because of the current state of the ledger but that could be applied under some imagined ledger.

I suggest `temBAD_SEQUENCE` instead.